src/App.tsx

```diff
@@ -15,47 +16,118 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+  const sign = isReversed === false ? 1 : -1;
+
+  visibleGoods.sort((good1, good2) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return sign * (good1.localeCompare(good2));
+      case SortType.LENGTH:
+        return sign * (good1.length - good2.length
+          + 0.1 * (good1.localeCompare(good2) > 0 ? 1 : -1));
+      case SortType.NONE:
+        return sign * (goodsFromServer.indexOf(good1)
+                  - goodsFromServer.indexOf(good2));
+      default:
+        return 0;
+    }
+  });
+
+  return visibleGoods;
+}
```

Comment:
Try to do it the same way as in the previous react_list-of-goods (without the `sign`)